server/api/documents.js

```diff
@@ -94,4 +94,40 @@ router.get(
   }
 );
 
+router.delete(
+  `/:id`,
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    const documentId = req.params.id;
+    const userId = req.user.id;
+
+    if (req.user === false) {
+      return res.send("Unauthorised");
+    } else {
+      docsDb
+        .checkDocumentStatus(documentId)
+        .then(status => {
+          if (status !== "Waiting") {
+            return res.status(400).send("Error");
```

Comment:
This works but with an error in the server side: `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`. Because you're in the middle of a promise chain here so you cannot return res.status from here. You'll need to throw an exception, which will be caught below in the catch and the error will be returned to the frontend.

Reply:
ok